src/libraries/System.Configuration.ConfigurationManager/Directory.Build.props

```diff
@@ -3,6 +3,7 @@
   <PropertyGroup>
     <StrongNameKeyId>Open</StrongNameKeyId>
     <IncludePlatformAttributes>true</IncludePlatformAttributes>
+    <UnsupportedOSPlatforms>browser</UnsupportedOSPlatforms>
```

Comment:
@buyaa-n @joperezr @krwq @steveharter @jeffhandley can you please double check if this is right? The platform analyzer warned for browser when I added the .NETCoreApp configuration.

Reply:
I am not sure if we can mark the entire assembly cc @marek-safar

Reply:
here's the error output without the setting:

```
C:\git\runtime2\src\libraries\System.CodeDom\src\System\CodeDom\Compiler\Executor.cs(62,21): error CA1416: This call site is reachable on all platforms. 'ProcessStartInfo.WorkingDirectory' is unsupported on: 'browser'. [C:\git\runtime2\src\libraries\System.CodeDom\src\System.CodeDom.csproj]
C:\git\runtime2\src\libraries\System.CodeDom\src\System\CodeDom\Compiler\Executor.cs(67,36): error CA1416: This call site is reachable on all platforms. 'Process.Start(ProcessStartInfo)' is unsupported on: 'tvos', 'browser', 'maccatalyst', 'ios'. [C:\git\runtime2\src\libraries\System.CodeDom\src\System.CodeDom.csproj]
C:\git\runtime2\src\libraries\System.CodeDom\src\System\CodeDom\Compiler\Executor.cs(85,21): error CA1416: This call site is reachable on all platforms. 'Process.BeginErrorReadLine()' is unsupported on: 'browser'. [C:\git\runtime2\src\libraries\System.CodeDom\src\System.CodeDom.csproj]
C:\git\runtime2\src\libraries\System.CodeDom\src\System\CodeDom\Compiler\Executor.cs(84,21): error CA1416: This call site is reachable on all platforms. 'Process.BeginOutputReadLine()' is unsupported on: 'browser'. [C:\git\runtime2\src\libraries\System.CodeDom\src\System.CodeDom.csproj]
C:\git\runtime2\src\libraries\System.CodeDom\src\System\CodeDom\Compiler\Executor.cs(60,27): error CA1416: This call site is reachable on all platforms. 'ProcessStartInfo' is unsupported on: 'browser'. [C:\git\runtime2\src\libraries\System.CodeDom\src\System.CodeDom.csproj]
C:\git\runtime2\src\libraries\System.CodeDom\src\System\CodeDom\Compiler\Executor.cs(76,21): error CA1416: This call site is reachable on all platforms. 'Process.ErrorDataReceived' is unsupported on: 'browser'. [C:\git\runtime2\src\libraries\System.CodeDom\src\System.CodeDom.csproj]
C:\git\runtime2\src\libraries\System.CodeDom\src\System\CodeDom\Compiler\Executor.cs(93,21): error CA1416: This call site is reachable on all platforms. 'Process.WaitForExit()' is unsupported on: 'browser'. [C:\git\runtime2\src\libraries\System.CodeDom\src\System.CodeDom.csproj]
C:\git\runtime2\src\libraries\System.CodeDom\src\System\CodeDom\Compiler\Executor.cs(73,52): error CA1416: This call site is reachable on all platforms. 'DataReceivedEventArgs.Data' is unsupported on: 'browser'. [C:\git\runtime2\src\libraries\System.CodeDom\src\System.CodeDom.csproj]
C:\git\runtime2\src\libraries\System.CodeDom\src\System\CodeDom\Compiler\Executor.cs(78,29): error CA1416: This call site is reachable on all platforms. 'DataReceivedEventArgs.Data' is unsupported on: 'browser'. [C:\git\runtime2\src\libraries\System.CodeDom\src\System.CodeDom.csproj]
C:\git\runtime2\src\libraries\System.CodeDom\src\System\CodeDom\Compiler\Executor.cs(80,51): error CA1416: This call site is reachable on all platforms. 'DataReceivedEventArgs.Data' is unsupported on: 'browser'. [C:\git\runtime2\src\libraries\System.CodeDom\src\System.CodeDom.csproj]
C:\git\runtime2\src\libraries\System.CodeDom\src\System\CodeDom\Compiler\Executor.cs(64,21): error CA1416: This call site is reachable on all platforms. 'ProcessStartInfo.RedirectStandardError' is unsupported on: 'browser'. [C:\git\runtime2\src\libraries\System.CodeDom\src\System.CodeDom.csproj]
C:\git\runtime2\src\libraries\System.CodeDom\src\System\CodeDom\Compiler\Executor.cs(87,26): error CA1416: This call site is reachable on all platforms. 'Process.WaitForExit(int)' is unsupported on: 'browser'. [C:\git\runtime2\src\libraries\System.CodeDom\src\System.CodeDom.csproj]
C:\git\runtime2\src\libraries\System.CodeDom\src\System\CodeDom\Compiler\Executor.cs(69,21): error CA1416: This call site is reachable on all platforms. 'Process.OutputDataReceived' is unsupported on: 'browser'. [C:\git\runtime2\src\libraries\System.CodeDom\src\System.CodeDom.csproj]
C:\git\runtime2\src\libraries\System.CodeDom\src\System\CodeDom\Compiler\Executor.cs(95,28): error CA1416: This call site is reachable on all platforms. 'Process.ExitCode' is unsupported on: 'browser'. [C:\git\runtime2\src\libraries\System.CodeDom\src\System.CodeDom.csproj]
C:\git\runtime2\src\libraries\System.CodeDom\src\System\CodeDom\Compiler\Executor.cs(63,21): error CA1416: This call site is reachable on all platforms. 'ProcessStartInfo.RedirectStandardOutput' is unsupported on: 'browser'. [C:\git\runtime2\src\libraries\System.CodeDom\src\System.CodeDom.csproj]
C:\git\runtime2\src\libraries\System.CodeDom\src\System\CodeDom\Compiler\Executor.cs(71,29): error CA1416: This call site is reachable on all platforms. 'DataReceivedEventArgs.Data' is unsupported on: 'browser'. [C:\git\runtime2\src\libraries\System.CodeDom\src\System.CodeDom.csproj]
```

Reply:
@buyaa-n I assume we don't run the platform analyzer on .NETStandard builds by default?

Reply:
Yes, by default run only on .Net 5.0 or above

Reply:
Why not for netstandard2.0 and netstandard2.1?

Reply:
To avoid false positives as mentioned [here](https://github.com/dotnet/roslyn-analyzers/issues/4091), but probably we should enable it in runtime repo

Reply:
Hitting this issues shows that some our shipping assemblies won't be compatible on all target (frameworks). We should either include .NETStandard assemblies as well or consider adding a $(NetCoreAppCurrent) configuration to all our libraries. When I checked yesterday I found ~40 libs that don't have a .NETCoreApp configuration but a .NETStandard one.

Reply:
> I am not sure if we can mark the entire assembly

I think we could.

> here's the error output without the setting

I'm surprised you get the errors for the browser only. The API is not supported on other targets as well (e.g. iOS)

Reply:
Doesn't System.CodeDom call `csc`? I don't know what it does on .NET Core but I presume that limits its reach to places that have the full SDK installed as well, as I'm not sure whether csc is installed when you just install the runtime.